src/EventTickets/Routes/DeleteEventTicketType.php

```diff
@@ -49,36 +55,25 @@ public function registerRoute()
     /**
      * @since 3.6.0
      *
+     * @return WP_REST_Response|WP_Error
      * @throws Exception
      */
-    public function handleRequest(WP_REST_Request $request): WP_REST_Response
+    public function handleRequest(WP_REST_Request $request)
     {
         $ticketType = EventTicketType::find($request->get_param('ticket_type_id'));
 
         $salesCount = $ticketType->eventTickets()->count();
 
         if ($salesCount > 0) {
-            return new WP_REST_Response([
-                'message' => __('This ticket type has been sold and cannot be deleted.', 'give'),
-            ], 400);
+            return new WP_Error(
```

Comment:
What is the benefit in changing this from a `WP_REST_Response` to a `WP_Error`?

Reply:
Interesting point. In that case, should this check be moved to the attribute `validate_callback`?

Reply:
I do agree. Moved it to the `validation_callback` while keeping the custom message.